### DISCUSSION TITLE
imageResizeTargetWidth makes smaller image larger to hit width. Attempting to only make large images smaller.

### DISCUSSION THREAD

**Discussion by ralphmorris** [filamentphp/filament#11603] [Help]
### Package

Form builder

### Package Version

3.0-stable

### How can we help you?

Hey,

I have the following code that is attempting to upload an image and resize it to make it smaller to only the required width and keeping the height dynamic, if the image is unnecessarily large.
```
Forms\Components\FileUpload::make('image')
    ->directory('cms')
    ->image()
    ->imageEditor()
    ->imageEditorAspectRatios([
        null,
        '16:9',
        '9:16',
        '4:3',
        '3:4',
        '1:1',
    ])
    ->imageResizeMode('contain')
    ->imageResizeTargetWidth('1288')
```
However, it also seems to make smaller images larger to hit the 1288px width. In previous projects I've been able to use intervention image to reduce an image by only it's width but keeping the aspect ratio. Is there a way to do this in Filament? I think it would be a fairly common use case?

Thanks!

**Comment by dberri** [comment]
You can use `->imageResizeUpscale(false)` on the FileUpload component to prevent that.
